### PR TITLE
Remove puppet and randomness from the test_positive_end_to_end

### PR DESCRIPTION
**Pull request by jaryn**
Random walk that this test was using was making it behaving flaky. This patch makes the test
parametrized to do thorough testing of the product rather than using a random walk.

Note that this patch increases the running time significantly as all of the defined paths are being exhamined.

To mitigate this we need to reduce the number of the testable components in the product. I filled a bug about this:
https://bugzilla.redhat.com/show_bug.cgi?id=2097309

Also remove Puppet.

**Note that merging this will increase the time taken by the tests by about an hour.**

### CODE REVIEW

**Comment by omkarkhatavkar** [comment]
Can one of the admins verify this patch?

File: tests/foreman/ui/test_bookmarks.py [hunk 1]
```diff
@@ -30,53 +28,45 @@
...
-                entity_setup(organization=[module_org], location=[module_location]).create()
-    return ui_entities
-
-
-@pytest.fixture()
-def random_entity(ui_entities):
-    """Returns one random entity from list of available UI entities"""
-    return ui_entities[random.randint(0, len(ui_entities) - 1)]
+            issues = {is_open(issue) for issue in skip}
+            pytest.skip(f'There is an open issue with this test: {issues}')
```
Comment (jyejare) [inline tests/foreman/ui/test_bookmarks.py:48]:
```suggestion
            pytest.skip(f'There is an open issue with entity: {issues}')
```
??

**Review by jyejare — APPROVED** [review]
ACK with minor nitpick comment !

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_bookmarks.py::test_positive_end_to_end

File: robottelo/constants/__init__.py [hunk 2]
```diff
@@ -1865,8 +1865,6 @@
         'setup': entities.ConfigGroup,
         'skip_for_ui': True,
     },
```
Comment (vsedmik) [inline robottelo/constants/__init__.py:1867]: When removing the puppet-related bookmarks, you might want to remove the ConfigGroup too. It is also puppet-related and fails on default 6.11 setup.

File: tests/foreman/ui/test_bookmarks.py [hunk 3]
```diff
@@ -30,53 +28,45 @@
...
-    return ui_entities[random.randint(0, len(ui_entities) - 1)]
+            issues = {is_open(issue) for issue in skip}
+            pytest.skip(f'There is/are an open issue(s) with this entity: {issues}')
+        # entities with 1 organization and location
+        if entity_name in ('Host',):
+            entity_setup(organization=module_org, location=module_location).create()
+        # entities with no organizations and locations
+        elif entity_name in (
+            'ComputeProfile',
+            'ConfigGroup',
```
Comment (vsedmik) [inline tests/foreman/ui/test_bookmarks.py:55]: Also here.

File: robottelo/constants/__init__.py [hunk 4]
```diff
@@ -1804,11 +1804,11 @@
 BOOKMARK_ENTITIES = [
     {'name': 'ActivationKey', 'controller': 'katello_activation_keys'},
     {'name': 'Dashboard', 'controller': 'dashboard', 'skip_for_ui': True},
-    {'name': 'Fact', 'controller': 'fact_values', 'skip_for_ui': True},
     {'name': 'Audit', 'controller': 'audits', 'skip_for_ui': True},
     {'name': 'Report', 'controller': 'config_reports', 'skip_for_ui': True},
     {'name': 'Task', 'controller': 'foreman_tasks_tasks', 'skip_for_ui': True},
-    {'name': 'Subscriptions', 'controller': 'katello_subscriptions', 'skip_for_ui': True},
+    # TODO Load manifest for the test_positive_end_to_end from the ui/test_bookmarks.py
+    # {'name': 'Subscriptions', 'controller': 'subscriptions', 'skip_for_ui': True},
```
Comment (pondrejk) [inline robottelo/constants/__init__.py:1811]: @jaryn did you intend to leave these lines in as comments?

Comment (jaryn) [reply robottelo/constants/__init__.py:1811]: I guess I should rather create a Jira ticket.

File: tests/foreman/ui/test_bookmarks.py [hunk 5]
```diff
@@ -30,53 +28,44 @@
...
-                entity_setup(organization=[module_org], location=[module_location]).create()
-    return ui_entities
-
-
-@pytest.fixture()
-def random_entity(ui_entities):
-    """Returns one random entity from list of available UI entities"""
-    return ui_entities[random.randint(0, len(ui_entities) - 1)]
+            issues = {is_open(issue) for issue in skip}
+            pytest.skip(f'There is/are an open issue(s) with this entity: {issues}')
```
Comment (mshriver) [inline tests/foreman/ui/test_bookmarks.py:48]: Significant improvement to the test coverage and testing pattern here, thanks @jaryn

File: tests/foreman/ui/test_bookmarks.py [hunk 6]
```diff
@@ -30,53 +28,44 @@
...
+        # entities with 1 organization and location
+        if entity_name in ('Host',):
+            entity_setup(organization=module_org, location=module_location).create()
+        # entities with no organizations and locations
+        elif entity_name in (
```
Comment (mshriver) [inline tests/foreman/ui/test_bookmarks.py:53]: Might be nice to put this in the datastructure for the entities, so the information about module/org context isn't built into the logic here.

**Comment by mshriver** [comment]
@jaryn thanks for including data about the increase to runtime - I think that's reasonable given the improved test coverage and consistency this provides. It's also a good case for xdist worker count linearly reducing runtime by going from 2 to 4 workers.

I'd like to see your response to my comment about the pytest.skip message and issue context before merging.

**Comment by mshriver** [comment]
@jaryn in the future please open PRs from branches on your fork, instead of pushing a fork to the main repo.

File: tests/foreman/ui/test_bookmarks.py [hunk 7]
```diff
@@ -30,53 +28,44 @@
...
-            else:
-                entity_setup(organization=[module_org], location=[module_location]).create()
-    return ui_entities
-
-
-@pytest.fixture()
-def random_entity(ui_entities):
-    """Returns one random entity from list of available UI entities"""
-    return ui_entities[random.randint(0, len(ui_entities) - 1)]
+            issues = {is_open(issue) for issue in skip}
```
[the remark this answers is not in the document]
Comment (jaryn) [reply tests/foreman/ui/test_bookmarks.py]: Done.

File: tests/foreman/ui/test_bookmarks.py (hunk 5 above)
Comment (jaryn) [reply tests/foreman/ui/test_bookmarks.py]: Thanks mike. Just a little change I am glad you see this as an improvement even though the test runtime is increased.

**Comment by jaryn** [comment]
@mshriver sorry for the inconvenience with the branches I didn't mean to do that. I've executed this to mitigate this in the future.
```
git remote set-url --push origin git@github.com:jaryn/robottelo.git
```

File: tests/foreman/ui/test_bookmarks.py (hunk 6 above)
Comment (jaryn) [reply tests/foreman/ui/test_bookmarks.py:53]: This is true. I would like to modify the data-structure. What is more important for us now is to make all the tests passing, isn't it? Sorry but I have to postpone this request bit towards the end of the universe with hope there will be time and me wise enough to do it lean and neat way. (:

Comment (mshriver) [reply tests/foreman/ui/test_bookmarks.py:53]: Definitely fine to not include that change in this PR!

**Review by mshriver — APPROVED** [review]

**Pull request merged**